src/Core/Auth/Models/Data/SsoConfigurationData.cs

```diff
@@ -22,7 +22,25 @@ public string Serialize()
 
     public SsoType ConfigType { get; set; }
 
-    public bool KeyConnectorEnabled { get; set; }
+    public MemberDecryptionType MemberDecryptionType { get; set; }
+
+    /// <summary>
+    /// Legacy property to determine if KeyConnector was enabled.
+    /// Kept for backwards compatibility with old configs that will not have
+    /// the new MemberDecryptionType when deserialized from the database.
+    /// </summary>
+    [Obsolete("Use MemberDecryptionType instead")]
+    public bool KeyConnectorEnabled
+    {
+        get => MemberDecryptionType == MemberDecryptionType.KeyConnector;
+        set
+        {
+            if (value)
+            {
+                MemberDecryptionType = MemberDecryptionType.KeyConnector;
+            }
```

Comment:
Maybe we should add an else statement and explicitly set `MemberDecryptionType = MemberDecryptionType.MasterPassword`. It's now being set to that because it's the default value

Reply:
My concern with adding the `else` statement was that the old `keyConnectorEnabled` field will always be present in the saved JSON, `true` or `false`. Which means, if someone were to enable TDE, `keyConnectorEnabled` would be set to `false` and the `else` would overwrite the explicit TDE option in `MemberDecryptionType` during deserialization from the database.

Reply:
Makes sense! Thanks